static/app/views/settings/organizationIntegrations/sentryAppDetailedView.tsx

```diff
@@ -20,188 +22,234 @@ import type {
 import {SplitInstallationIdModal} from './SplitInstallationIdModal';
 
-type State = {
-  appInstalls: SentryAppInstallation[];
-  featureData: IntegrationFeature[];
-  sentryApp: SentryApp;
-};
-
-type Tab = AbstractIntegrationDetailedView['state']['tab'];
-
-class SentryAppDetailedView extends AbstractIntegrationDetailedView<
-  AbstractIntegrationDetailedView['props'],
-  State & AbstractIntegrationDetailedView['state']
-> {
-  tabs: Tab[] = ['overview'];
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {
-      organization,
-      params: {integrationSlug},
-    } = this.props;
-    return [
-      ['sentryApp', `/sentry-apps/${integrationSlug}/`],
-      ['featureData', `/sentry-apps/${integrationSlug}/features/`],
-      ['appInstalls', `/organizations/${organization.slug}/sentry-app-installations/`],
-    ];
-  }
-
-  onLoadAllEndpointsSuccess() {
-    const {
-      organization,
-      params: {integrationSlug},
-      router,
-    } = this.props;
-
-    // redirect for internal integrations
-    if (this.sentryApp.status === 'internal') {
-      router.push(
-        `/settings/${organization.slug}/developer-settings/${integrationSlug}/`
-      );
-      return;
-    }
-
-    super.onLoadAllEndpointsSuccess();
-    recordInteraction(integrationSlug, 'sentry_app_viewed');
-  }
-
-  get integrationType() {
-    return 'sentry_app' as const;
-  }
-
-  get sentryApp() {
-    return this.state.sentryApp;
-  }
-
-  get description() {
-    return this.state.sentryApp.overview || '';
-  }
-
-  get author() {
-    return this.sentryApp.author;
-  }
+function makeSentryAppInstallationsQueryKey({orgSlug}: {orgSlug: string}): ApiQueryKey {
+  return [`/organizations/${orgSlug}/sentry-app-installations/`];
+}
 
-  get resourceLinks() {
-    // only show links for published sentry apps
-    if (this.sentryApp.status !== 'published') {
+export default function SentryAppDetailedView() {
+  const tabs: Tab[] = ['overview'];
+  const api = useApi();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const organization = useOrganization();
+  const {integrationSlug} = useParams<{integrationSlug: string}>();
+
+  const {data: sentryApp, isPending: isSentryAppPending} = useApiQuery<SentryApp>(
+    [`/sentry-apps/${integrationSlug}/`],
+    {staleTime: Infinity, retry: false}
+  );
+
+  const {data: featureData = [], isPending: isFeatureDataPending} = useApiQuery<
+    IntegrationFeature[]
+  >([`/sentry-apps/${integrationSlug}/features/`], {staleTime: Infinity, retry: false});
+
+  const {data: appInstalls = [], isPending: isAppInstallsPending} = useApiQuery<
+    SentryAppInstallation[]
+  >(makeSentryAppInstallationsQueryKey({orgSlug: organization.slug}), {
+    staleTime: Infinity,
+    retry: false,
+  });
+
+  const integrationType = 'sentry_app';
+  const description = useMemo(() => sentryApp?.overview || '', [sentryApp?.overview]);
```

Comment:
Do you need useMemo here? Seems like an uneeded optimization no? Nothing expensive is happening here and nothing unstable across renders needs to be made stable

Reply:
good catch, ive been working on these in parallel and malachi pointed out the same thing in another PR. I just did the same thing for every getter but these are unecessary :+1:

Reply:
Yeah I think most of them don't need useMemo 👍